qadence_libs/qinfo_tools/qfi.py

```diff
@@ -0,0 +1,166 @@
+from __future__ import annotations
+
+from typing import Iterable
+
+import torch
+from qadence import Overlap
+from qadence.blocks import parameters, primitive_blocks
+from qadence.circuit import QuantumCircuit
+from qadence.types import BackendName, DiffMode, OverlapMethod
+from torch import Tensor
+
+from qadence_libs.qinfo_tools.spsa import spsa_2gradient_step
+from qadence_libs.qinfo_tools.utils import hessian
+
+
+def _symsqrt(A: Tensor) -> Tensor:
+    """Computes the square root of a Symmetric or Hermitian positive definite matrix.
+
+    or batch of matrices.
+
+    Code from https://github.com/pytorch/pytorch/issues/25481#issuecomment-1032789228
```

Comment:
Is this code stolen form that issue ?

Reply:
Pretty much. Is this okay? There's not really any functionality in torch to compute the square of a matrix so this was what I found. It was posted by a torch developer in the issue.

Reply:
I did do some checks and it worked as intended. I could also add a test in `tests` for the function to be sure

Reply:
I added a test for this.